File: csr_magic_frontend/src/types/event.ts
export type EventType = 'OFFLINE' | 'ONLINE' | 'HYBRID';

export interface Event {
  id: number;
  name: string;
  description: string | null;
  type: EventType | null;
  startDate: string | null;
  endDate: string | null;
  coverImage: string | null;
  visible: boolean;
  createdAt: string;
  updatedAt: string | null;
}

export interface CreateEventRequest {
  name: string;
  description?: string;
  type?: EventType;
  startDate?: string;
  endDate?: string;
  coverImage?: string;
  visible?: boolean;
}

export interface UpdateEventRequest {
  name?: string;
  description?: string;
  type?: EventType;
  startDate?: string;
  endDate?: string;
  coverImage?: string;
  visible?: boolean;
}

export interface EventListParams {
  page?: number;
  size?: number;
  keyword?: string;
}

File: csr_magic_backend/src/main/java/com/csr/event/entity/Event.java
package com.csr.event.entity;

import jakarta.persistence.*;
import java.time.Instant;

@Entity
@Table(name = "event")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 20)
    private String type;

    private Instant startDate;

    private Instant endDate;

    @Column(columnDefinition = "TEXT")
    private String coverImage;

    @Column(nullable = false)
    private Boolean visible = true;

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = Instant.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = Instant.now();
    }

    // Getters and Setters

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Instant getStartDate() {
        return startDate;
    }

    public void setStartDate(Instant startDate) {
        this.startDate = startDate;
    }

    public Instant getEndDate() {
        return endDate;
    }

    public void setEndDate(Instant endDate) {
        this.endDate = endDate;
    }

    public String getCoverImage() {
        return coverImage;
    }

    public void setCoverImage(String coverImage) {
        this.coverImage = coverImage;
    }

    public Boolean getVisible() {
        return visible;
    }

    public void setVisible(Boolean visible) {
        this.visible = visible;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }
}
